### PR TITLE
Add set default button to address list to allow customer to change default address.

### PR DESCRIPTION
**Pull request by djtimca**
Add a button to the address list allowing a customer to set their default address in the system.

### CODE REVIEW

File: components/addressbook/list.blade.php [hunk 1]
```diff
@@ -10,6 +10,10 @@ class="list-group-item {{ ($customer->address_id == $address->address_id) ? 'lis
                         class="btn btn-outline-default"
                         href="{{ site_url('account/address', ['addressId' => $address->address_id]) }}"
                     >@lang('igniter.user::default.account.text_edit')</a>
+                    <a
+                        class="btn btn-outline-default"
+                        href="{{ site_url('account/address', ['defaultAddressId' => $address->address_id]) }}"
+                    >@lang('igniter.user::default.account.text_set_default')</a>
```
Comment (ryanmitchell) [inline components/addressbook/list.blade.php:16]: should be igniter.user::default.text_set_default

File: components/AddressBook.php [hunk 2]
```diff
@@ -32,6 +33,27 @@ public function onLoadAddForm()
         return ['#address-book' => $this->renderPartial('@form')];
     }
 
+    public function setDefaultAddress()
```
Comment (ryanmitchell) [inline components/AddressBook.php:36]: call it onSetDefaultAddress() to follow the conventions

Comment (ryanmitchell) [reply components/AddressBook.php:36]: Ignore this.

File: components/addressbook/list.blade.php (hunk 1 above)
Comment (ryanmitchell) [reply components/addressbook/list.blade.php:16]: Also wrap the link in an @if ($customer->address_id != $address->address_id) so it only shows if its not the default.

**Comment by ryanmitchell** [comment]
So you could do something like:

```
    protected function getAddress()
    {
        if (!is_numeric($addressIdParam = $this->param('addressId')))
            return null;

        $customer = Auth::customer();
        $address = Addresses_model::find($addressIdParam);
        if (!$customer OR $address->customer_id != $customer->customer_id)
            return null;
            
        if (\Request::get('setDefault'))
            $this->setDefaultAddress(￼);

        return $address;
    }
    
    public function setDefaultAddress($addressId)
    {
        $address = Addresses_model::find($addressId);

        $customer->address_id = $defaultAddressId;
        $customer->save();

        flash()->success(lang('igniter.user::default.account.alert_updated_success'))->now();

        return null;
    }
```

```
                    <a
                        class="btn btn-outline-default"
                        href="<?= site_url('account/address', ['addressId' => $address->address_id]); ?>"
                    >Edit</a>
                    <a
                        class="btn btn-outline-default"
                        href="<?= site_url('account/address', ['addressId' => $address->address_id]); ?>?setDefault=1"
                    >Default</a>
```

Notice the url param, and you're leveraging the address checking already in place in getAddress().

@sampoyigi might have a different approach in mind.

**Comment by sampoyigi** [comment]
Thank you for your contributions @djtimca!

I'd go with @ryanmitchell approach, except I'll do `\Request::get('setDefault') == '1'` instead ;)

File: components/AddressBook.php (hunk 2 above)
Comment (djtimca) [reply components/AddressBook.php]: I didn't change it - left as per comment below. Let me know if you want it changed.

**Review by sampoyigi — CHANGES_REQUESTED** [review]

File: components/AddressBook.php [hunk 3]
```diff
@@ -86,9 +86,22 @@ protected function getAddress()
         if (!$customer OR $address->customer_id != $customer->customer_id)
             return null;
 
+        if (\Request::get('setDefault') == '1')
+            $this->setDefaultAddress($customer, $addressIdParam);
+
```
[the remark this answers is not in the document]
Comment (djtimca) [reply components/AddressBook.php]: OK moved things around and I think I have what you were looking for. Tested behaviour is the redirect with new default address set. I excluded any of the verification that is in the getAddress function but wondering if I should replicate that somehow for security. Low risk given it is just setting default address.

**Review by sampoyigi — APPROVED** [review]

**Comment by sampoyigi** [comment]
Thank you!

**Pull request merged**